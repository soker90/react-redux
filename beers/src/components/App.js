import React, { Component } from 'react'
import { BrowserRouter as Router, Route,Link } from 'react-router-dom'
import BeerPage from '../containers/BeerPage'
import ModalPage from '../containers/ModalPage'
import About from '../components/About'
import logo from '../logo.svg'
import './App.css'
import CoinPage from '../containers/CoinPage'

class App extends Component {
	openModal() {
		let modalData = {
			title: 'This is the modal title!',
			body: 'Lorem Ipsum es simplemente el texto de relleno de las imprentas y archivos de texto. Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500.'
		}
		this.props.modalAction.openModal(modalData)
	}

  render() {
    return (
    <Router>
      <div className="app">
        {this.props.modal.open && <ModalPage />}
        <div className="app-header">
          <img src={logo} className="app-logo" alt="logo" />
          <h2>Welcome to BeersApp</h2>
        </div>
        <nav>
          <ul>
            <li><Link to={'/beers'}>Beers</Link></li>
            <li><Link to={'/coins'}>Coins</Link></li>
            <li><Link to={'/about'}>About</Link></li>
          </ul>
        </nav>
        <button onClick={() => this.openModal()}>Open Modal</button>

        {
          (() => {
            if (process.env.NODE_ENV === 'development') {
              const DevTools = require('../containers/DevTools').default
              return <DevTools />
            }
          })()
        }

        <Route path="/about" component={About}/>
        <Route path="/beers" component={BeerPage}/>
        <Route path="/coins" component={CoinPage}/>
      </div>
    </Router>
    )
  }
}

export default App
