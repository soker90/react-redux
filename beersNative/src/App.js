import React, { Component } from 'react';
import {
	Platform,
	StyleSheet,
	Text,
	View
} from 'react-native'

import configureStore from './store/configureStore'
import { Provider } from 'react-redux'
import CoinPage from './containers/CoinPage'
import About from './components/About'
import BeerPage from './containers/BeerPage'
import { MemoryRouter as Router, Route, Link } from 'react-router-native'

const store = configureStore()

export default class App extends Component {
	render() {
		return (
			<Provider store={store}>
				<View style={styles.container}>
					<Router>
						<View className="app">
							<View className="app-header">
								<Link to={'/beers'}><Text>Beers</Text></Link>
								<Link to={'/coins'}><Text>Coins</Text></Link>
								<Link to={'/about'}><Text>About</Text></Link>
							</View>


							<Route path="/about" component={About}/>
							<Route path="/beers" component={BeerPage}/>
							<Route path="/coins" component={CoinPage}/>
					</View>
				</Router>
			</View>
		</Provider>


			/*<View style={styles.container}>
				/*<Router>
					<View className="app">
						<View className="app-header">
							<Text>Welcome to BeersApp</Text>
						</View>
						/*<nav>
							<ul>
								<li><Link to={'/beers'}>Beers</Link></li>
								<li><Link to={'/coins'}>Coins</Link></li>
								<li><Link to={'/about'}>About</Link></li>
							</ul>
						</nav>
						<button onClick={() => this.openModal()}>Open Modal</button>



						<Route path="/about" component={About}/>
						<Route path="/beers" component={BeerPage}/>
						<Route path="/coins" component={CoinPage}/>*/
						/*</View>
					</Router>
				</View>*/
			//</Provider>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#F5FCFF',
	},
	welcome: {
		fontSize: 20,
		textAlign: 'center',
		margin: 10,
	},
	instructions: {
		textAlign: 'center',
		color: '#333333',
		marginBottom: 5,
	},
});
