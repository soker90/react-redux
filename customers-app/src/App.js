import React, { Component } from 'react'
import './App.css'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import HomeContainer from './containers/HomeContainer'
import CustomersContainer from './containers/CustomersContainer'
import CustomerContainer from './containers/CustomerContainer'

class App extends Component {
	renderCustomerNewContainer = () => <h1>Customer New Container</h1>

	render() {
		return (
			<Router>
				<div>
					<Route exact path="/" component={HomeContainer}/>
					<Route exact path="/customers" component={CustomersContainer}/>
					<Switch>
						<Route path="/customers/new" component={this.renderCustomerNewContainer()}/>
						<Route path="/customers/:dni" render={props => <CustomerContainer dni={props.match.params.dni} />}/>
					</Switch>
				</div>
			</Router>
		)
	}
}



export default App
