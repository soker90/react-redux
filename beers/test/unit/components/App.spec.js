// App.spec.js
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../../../src/components/App'

describe('App component', () => {
  test('render App component without crashing', () => {
    const modal = {
      open:false
    }

    const tree = renderer.create(
      <App modal={modal}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

})
