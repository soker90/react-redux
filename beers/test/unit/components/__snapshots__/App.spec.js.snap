exports[`App component render App component without crashing 1`] = `
<div
  className="app">
  <div
    className="app-header">
    <img
      alt="logo"
      className="app-logo"
      src="logo.svg" />
    <h2>
      Welcome to BeersApp
    </h2>
  </div>
  <nav>
    <ul>
      <li>
        <a
          href="/beers"
          onClick={[Function]}>
          Beers
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}>
          About
        </a>
      </li>
    </ul>
  </nav>
  <button
    onClick={[Function]}>
    Open Modal
  </button>
</div>
`;
